File: repo-root/groups/group-01-manage-field/code/managefield/src/main/java/com/jad/efreifive/managefield/controller/ReferenceDataController.java
package com.jad.efreifive.managefield.controller;

import com.jad.efreifive.managefield.dto.FieldStatusDto;
import com.jad.efreifive.managefield.dto.ReservationStatusDto;
import com.jad.efreifive.managefield.mapper.ReferenceDataMapper;
import com.jad.efreifive.managefield.service.ReferenceDataService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/v1")
@RequiredArgsConstructor
public class ReferenceDataController {

    private final ReferenceDataService referenceDataService;
    private final ReferenceDataMapper referenceDataMapper;

    @GetMapping("/field-statuses")
    public List<FieldStatusDto> listFieldStatuses() {
        return referenceDataService.listFieldStatuses().stream()
            .map(referenceDataMapper::toDto)
            .toList();
    }

    @GetMapping("/reservation-statuses")
    public List<ReservationStatusDto> listReservationStatuses() {
        return referenceDataService.listReservationStatuses().stream()
            .map(referenceDataMapper::toDto)
            .toList();
    }
}

File: repo-root/groups/group-01-manage-field/code/manage-field/src/main/kotlin/fr/efrei/managefield/controller/ReferenceDataController.kt
package fr.efrei.managefield.controller

import fr.efrei.managefield.controller.dto.response.FieldStatusResponseDto
import fr.efrei.managefield.controller.dto.response.ReservationStatusResponseDto
import fr.efrei.managefield.mapper.ReferenceDataApiMapper
import fr.efrei.managefield.service.ReferenceDataService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * Exposes HTTP endpoints for status reference data.
 */
@RestController
class ReferenceDataController(
    private val referenceDataService: ReferenceDataService,
    private val referenceDataApiMapper: ReferenceDataApiMapper
) {
    @GetMapping("/v1/field-statuses")
    fun listFieldStatuses(): List<FieldStatusResponseDto> {
        return referenceDataApiMapper.toFieldStatusResponses(referenceDataService.listFieldStatuses())
    }

    @GetMapping("/v1/reservation-statuses")
    fun listReservationStatuses(): List<ReservationStatusResponseDto> {
        return referenceDataApiMapper.toReservationStatusResponses(referenceDataService.listReservationStatuses())
    }
}
